### PR TITLE
React sign out

### PR DESCRIPTION
**Pull request by powerful23**
For issue #180 

Refreshing federated jwt token on need.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 1]
```diff
@@ -46,11 +48,51 @@ export default class Greetings extends AuthPiece {
     }
 
     signOut() {
+        this.googleSignOut();
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Greetings.jsx]: We do some sort of check here first before calling these so we don't run into unexpected exceptions with the specific frameworks. Perhaps we should abstract this out into the Auth.signOut() method as well, so the error returned in catch() could be related to federated.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 2]
```diff
@@ -46,11 +48,51 @@ export default class Greetings extends AuthPiece {
...
+        this.facebookSignOut();
         Auth.signOut()
             .then(() => this.changeState('signedOut'))
             .catch(err => { logger.error(err); this.error(err); });
     }
 
+    googleSignOut() {
+        const auth2 = window.gapi && window.gapi.auth2? window.gapi.auth2 : null;
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Greetings.jsx]: update here to not include int in the string name, use something like authGoogle etc.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 3]
```diff
@@ -46,11 +48,51 @@ export default class Greetings extends AuthPiece {
...
+                return Promise.resolve(null);
+            }
+
+            logger.debug('google signing out');
+            return googleAuth.signOut();
+        });
+    }
+
+    facebookSignOut() {
+        const fb = window.FB;
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Greetings.jsx]: check if window.FB exists first

File: packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx [hunk 4]
```diff
@@ -33,7 +34,7 @@ export default function withGoogle(Comp) {
             };
 
             const { onStateChange } = this.props;
-            return Auth.federatedSignIn('google', { token: id_token, expires_at }, user)
+            return Auth.federatedSignIn('google', { token: id_token, expires_at, refreshing: false }, user)
                 .then(crednetials => {
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx]: typo 'crednetials ' = credentials

File: packages/aws-amplify-react/src/Auth/Provider/withFacebook.jsx [hunk 5]
```diff
@@ -46,6 +46,7 @@ export default function withFacebook(Comp) {
             if (!accessToken) {
                 return;
             }
+            const that = this;
```
Comment (richardzcode) [inline packages/aws-amplify-react/src/Auth/Provider/withFacebook.jsx]: remove

File: packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx [hunk 6]
```diff
@@ -32,6 +32,7 @@ export default function withGoogle(Comp) {
                 name: profile.getName()
             };
 
+            const that = this;
```
Comment (richardzcode) [inline packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx]: remove

File: packages/aws-amplify-react/src/Auth/Greetings.jsx (hunk 1 above)
Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: Check is implemented in the beginning of those methods.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx (hunk 2 above)
Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: auth2 is a name given by Google.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx (hunk 3 above)
Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: Will be checked in the next line.

**Pull request merged**